src/nginx/status.cc

```diff
@@ -270,6 +276,34 @@ Status stats_json_per_process(const ngx_esp_process_stats_t &process_stats,
   return utils::ProtoToJson(status, json, utils::JsonOptions::OUTPUT_DEFAULTS);
 };
 
+ngx_int_t ngx_esp_update_rollout(std::shared_ptr<ApiManager> esp,
+                                 ngx_esp_process_stats_t *process_stat,
+                                 int index) {
+  ServiceConfigRolloutsInfo rollouts_info;
+  esp->GetServiceConfigRollouts(&rollouts_info);
+
+  ServiceConfigRolloutsProto service_config_rollouts_proto;
+  service_config_rollouts_proto.set_rollout_id(rollouts_info.rollout_id);
+  for (auto percentage : rollouts_info.percentages) {
+    (*service_config_rollouts_proto.mutable_percentages())[percentage.first] =
+        percentage.second;
+  }
+
+  std::string rollouts;
+  service_config_rollouts_proto.SerializeToString(&rollouts);
+
+  if (rollouts.length() <= kMaxServiceRolloutsInfoSize &&
+      rollouts.length() > 0 &&
+      strncmp(process_stat->esp_stats[index].rollouts, rollouts.c_str(),
+              rollouts.length()) != 0) {
+    process_stat->esp_stats[index].rollouts_length = rollouts.length();
+    strncpy(process_stat->esp_stats[index].rollouts, rollouts.c_str(),
```

Comment:
use `memcpy`, binary serialized protobuf might have NULL in the middle. Also consider check `ByteSize` of protobuf and directly use `SerializeToArray` instead of copying.